### ISSUE TITLE
Add progress parameter for ShineFX

### ISSUE THREAD

**Issue by orcomarcio** [phaserjs/phaser#6732]
The ShineFX it's cool looking, but without being able to control it it's hard to make it useful.
To try and make a single shine I even tried to add/remove the fx only when i need it, but it seems the progress it's tied to global time or something like that, so when I add it back in it doesn't start form the top.
Unless you want something shining constantly or you don't mind turning it on/off in an abrupt manner, is hard to use.
I'd like to control the progress myself, like shining once or moving the reflex back and forth (basically have a value i can tween as i fit).

**Comment by spayton** [comment]
yes, this would be useful, for each shine to optionally have its own progress, so you can add it to something and ensure it doesn't start midway

**Comment by BenjaminDRichards** [comment]
For reference, Phaser v4 removes the Shine effect, but we are planning examples for adding it to games that require it.

We looked at all the possible ways to style a shine, and decided that it would be more flexible if it were controlled by an arbitrary gradient. Then we looked at arbitrary gradients, and decided there were too many possibilities to include.

The solution will be something like rendering a shine gradient with a Shader, and applying it to the target using a Mask filter. This more open implementation will be easier to adjust.

I'm leaving this issue open until we've implemented an example, so we can link to it.

**Comment by BenjaminDRichards** [comment]
Update: Phaser 4 does not have a built-in Shine effect. However, there are two examples of shine effects in the `examples` repo (https://github.com/phaserjs/examples) built against Phaser 4.

The first is `game objects/shaders/gradient shine`. This demonstrates how to construct a shader from several pieces. It uses a bilinear gradient to create a shine effect, and a displacement filter to conform it to background contours. This is a very technical approach, suitable as the basis for more complex effects.

The second is `filters/custom shine filter` (and `filters/custom shine filter with progress control`). This is a dedicated shine effect. I added an optional progress control - it can still use game time, but you can also tween progress manually.

Both effects import files in the `public/assets/` folder. The textures are not licensed for use, but the code by us is fine, and could be copied into projects no problem.
